### ISSUE TITLE
gh_file_search.py results are not repeatable

### ISSUE THREAD

**Issue by hwine** [MoCo-GHE-Admin/github-scripts#72]
We may not be able to do anything, but I don't always get the same results running the same query. The changes in #71 got me far enough to feel comfortable that it is not a script logic error (one manifestation could have been explained if orgs were somehow skipped from a long list).

**STR**:
- `./gh_file_search.py -v -t 20 --query 'from ubuntu 22 filename:Dockerfile*' --note-archive -f mozilla{,-{it,iam,services,sre-deploy}}` Should return 2 hits, 1 each from:
  - `mozilla/ci-docker-bases`
  - `mozilla-services/socorro`

- `./gh_file_search.py -v -t 20 --query 'from ubuntu 22 filename:Dockerfile*' --note-archive -f $(cat all-orgs.txt)`
  - should report same 2 hits, usually only reports 1

**Comment by cknowles-admin** [comment]
Can you share the all-orgs.txt?  Not that I think it's the concern - but so I can verify I'm trying the same things.

**Comment by cknowles-moz** [comment]
So running the first command - I get the same results as you, those 2 responses.
Running the second command - using my version of the all orgs file - I've run it three times now (takes a bit thanks to GH's hidden rate limit) and each time I get the same 2 responses.  I am using the pre-PR code - but you mentioned that version expressed the same problem.  

So I can't reproduce the problem.

**Comment by cknowles-moz** [comment]
The 6th run was the one that reproduced the problem.  so, "you're not crazy".  

Trying to do some stress testing.  Wondering if there's an error in the underlying library.

**Comment by cknowles-admin** [comment]
In some of my tests, I got no(!) results. I'm now working on rewriting the search to use the API directly to see if THAT's where the problem lies.

**Comment by cknowles-moz** [comment]
Have rewritten to use the straight up API calls ... and wow, it's still flaking out on finding the mozilla version (which is the one that primarily failed.)  This version is better as it doesn't appear to be as impacted by secondary limits, but ick.  This still stinks.

**Comment by cknowles-moz** [comment]
so let's go SIMPLER still.  Straight up curl.  

`curl "https://api.github.com/search/code?q=from+ubuntu+22+filename%3ADockerfile%2A+org%3AMozilla+org%3AMozilla-services+org%3AMozilla-it+org%3AMozilla-iam&per_page=100&page=1"
{
  "total_count": 1, 
...
`

Note, that queries several orgs all at once.  And while it USUALLY returns 2 entries (our friends in mozilla, mozilla-services) you can see here that only 1 was returned.  So... I think we're just stuck here.

**Comment by hwine** [comment]
Yeah - that's why I said - "may not be fixable" -- I'm not surprised it's on GitHub's end.

Rather than beat our head's against a brick wall (that we didn't build), it might make more sense to wait for the code search API, and just have this be a "known bug".

**Comment by cknowles-moz** [comment]
So let's go simpler still - Curl

`curl "https://api.github.com/search/code?q=from+ubuntu+22+filename%3ADockerfile%2A+org%3AMozilla+org%3AMozilla-services+org%3AMozilla-it+org%3AMozilla-iam&per_page=100&page=1"`
`{
  "total_count": 1,
...`

The query hits against several orgs (verified in the UI that this returns what is expected) but you can see this response, it only got one.  Usually it's returning 2 - but every so often, 1.  

So I think we're into "this is not us."

**Comment by cknowles-moz** [comment]
As I was getting ready to file a ticket with support - decided to reproduce it one more time ... and I can't.  I've run the curl approximately 100 times, every time coming back with 2 results, running the script 3 times (working on a 4th) and getting 2 results each time.   I wonder if the "dumb node of the cluster" theory might be right, and they just kicked it.  Now I'm really not sure what to do.
